File: testdata/jennies/builders/map_of_disjunctions/JavaBuilder/map_of_disjunctions/LibraryPanelBuilder.java
package map_of_disjunctions;


public class LibraryPanelBuilder implements cog.Builder<LibraryPanel> {
    protected final LibraryPanel internal;
    
    public LibraryPanelBuilder() {
        this.internal = new LibraryPanel();
        this.internal.kind = "Library";
    }
    public LibraryPanelBuilder text(String text) {
        this.internal.text = text;
        return this;
    }
    public LibraryPanel build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/map_of_disjunctions/PHPBuilder/src/MapOfDisjunctions/LibraryPanelBuilder.php
<?php

namespace Grafana\Foundation\MapOfDisjunctions;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\MapOfDisjunctions\LibraryPanel>
 */
class LibraryPanelBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\MapOfDisjunctions\LibraryPanel $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\MapOfDisjunctions\LibraryPanel();
    $this->internal->kind = "Library";
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\MapOfDisjunctions\LibraryPanel
     */
    public function build()
    {
        return $this->internal;
    }

    public function text(string $text): static
    {
        $this->internal->text = $text;
    
        return $this;
    }

}
